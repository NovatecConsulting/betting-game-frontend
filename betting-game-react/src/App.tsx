import React from "react";
import { BrowserRouter as Router } from "react-router-dom";
import { routes, RoutesProvider, RouteSwitch } from "./routes";
import { AuthContext } from "./auth";

const App: React.FC = () => (
    <AuthContext>
        <RoutesProvider value={routes}>
            <Router basename={process.env.PUBLIC_URL}>
                <RouteSwitch />
            </Router>
        </RoutesProvider>
    </AuthContext>
);

export default App;
