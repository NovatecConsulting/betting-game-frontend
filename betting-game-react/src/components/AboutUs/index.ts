export { default as AboutUs } from "./AboutUs";
