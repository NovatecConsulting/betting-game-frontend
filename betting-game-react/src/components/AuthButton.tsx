import React from "react";
import { History } from 'history';
import {IAuthenticationSystem} from "../AuthenticationSystem";

interface AuthButtonProps {
    history: History,
    auth: IAuthenticationSystem
}

export const AuthButton = ({history, auth}: AuthButtonProps) => {
    let signout = () => {
        auth.signout(() => history.push("/"));
    };

    return auth.isAuthenticated
        ? (<p>
            Welcome {auth.userName}!
            <button onClick={signout}>
                Sign out
            </button>
        </p>)
        : (<p>You are not logged in.</p>);
}
