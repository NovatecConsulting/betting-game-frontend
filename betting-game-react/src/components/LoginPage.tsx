import React, {FormEvent, useState} from "react";
import {History} from "history"
import {IAuthenticationSystem} from "../AuthenticationSystem";

interface LoginPageProps {
    history: History,
    location: any,
    auth: IAuthenticationSystem
}

export const LoginPage = ({history, location, auth}: LoginPageProps) => {
    const fromPathName = location.state === undefined ? "/" : location.state.from.pathname;
    console.log(fromPathName);
    const [userName, setUserName] = useState("");
    const [password, setPassword] = useState("");

    let login = (e : FormEvent) => {
        e.preventDefault();
        auth.authenticate(
            userName,
            password,
            () => history.replace(fromPathName));
    };

    return (
        <div>
            <p>You must log in to view the page at {fromPathName}</p>
            <form onSubmit={e => login(e)}>
                <label>
                    Name:
                    <input type="text" name="name" required
                           value={userName} onChange={e => setUserName(e.target.value)}/>
                </label>
                <label>
                    Password:
                    <input type="password" name="password" required
                           value={password} onChange={e => setPassword(e.target.value)}/>
                </label>
                <input type="submit" value="Submit"/>
            </form>
        </div>
    );
}