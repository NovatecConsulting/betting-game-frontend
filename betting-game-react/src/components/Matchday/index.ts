export { default as Matchday } from "./Matchday";
