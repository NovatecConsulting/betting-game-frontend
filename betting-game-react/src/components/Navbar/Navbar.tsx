import React from "react";
import { Redirect, Route, RouteProps, Switch } from "react-router-dom";

import routes, { ABOUTUS, MATCHES, STANDINGS } from "../../config/routes";
import NavbarButton, { NavbarButtonProps } from "../NavbarButton/NavbarButton";
import NavbarLoginButton from "../NavbarLoginButton/NavbarLoginButton";
import NavbarLogo from "../NavbarLogo/NavbarLogo";

export const navbarButtons: NavbarButtonProps[] = [
    {
        buttonURL: MATCHES.path,
        buttonText: "Matches",
        buttonTestIdPrefix: "Matches",
    },
    {
        buttonURL: STANDINGS.path,
        buttonText: "Standings",
        buttonTestIdPrefix: "Standings",
    },
    {
        buttonURL: ABOUTUS.path,
        buttonText: "About us",
        buttonTestIdPrefix: "AboutUs",
    },
];

const Navbar: React.FC = () => {
    return (
        <span>
            <nav
                className="flex items-center justify-between flex-wrap border-b-2 border-blue-300 bg-blue-900 p-2"
                data-testid="Navbar"
            >
                <NavbarLogo />
                <div className="block flex-grow lg:flex lg:items-center lg:w-auto">
                    <div className="text-sm lg:flex-grow inline">
                        {navbarButtons.map(
                            (
                                navbarButton: NavbarButtonProps,
                                index: number
                            ) => (
                                <NavbarButton
                                    key={index}
                                    buttonText={navbarButton.buttonText}
                                    buttonURL={navbarButton.buttonURL}
                                    buttonTestIdPrefix={
                                        navbarButton.buttonTestIdPrefix
                                    }
                                ></NavbarButton>
                            )
                        )}
                    </div>
                </div>
                <NavbarLoginButton />
            </nav>
            <Switch>
                {routes.map((route: RouteProps, index: number) => (
                    <Route
                        data-testid="Route"
                        key={index}
                        exact
                        path={route.path}
                        component={route.component}
                    />
                ))}
                <Redirect to={MATCHES.path} />
            </Switch>
        </span>
    );
};

export default Navbar;
