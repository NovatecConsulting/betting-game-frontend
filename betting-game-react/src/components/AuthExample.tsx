import React, {useState} from 'react'
import {
    BrowserRouter as Router,
    Switch,
    Route,
    Link,
    useHistory,
    useLocation
} from "react-router-dom";
import {PrivateRoute} from "./PrivateRoute";
import {AuthButton} from "./AuthButton";
import {LoginPage} from "./LoginPage";
import {FakeAuthenticationSystem} from "../AuthenticationSystem";

export default function AuthExample() {
    return <Router> <Routing/> </Router>;
};

function Routing() {
    const history = useHistory();
    const location = useLocation();

    const [auth] = useState(new FakeAuthenticationSystem());

    return (
        <div>
            <AuthButton auth={auth} history={history}/>
            <ul>
                <li>
                    <Link to="/public">Public Page</Link>
                </li>
                <li>
                    <Link to="/protected">Protected Page</Link>
                </li>
            </ul>

            <Switch>
                <Route path="/public">
                    <PublicPage/>
                </Route>
                <Route path="/login">
                    <LoginPage history={history} location={location} auth={auth}/>
                </Route>
                <PrivateRoute path="/protected" isAuthenticated={auth.isAuthenticated}>
                    <ProtectedPage/>
                </PrivateRoute>
            </Switch>
        </div>
    )
}


const PublicPage = () => <h3>Public</h3>;

const ProtectedPage = () => <h3>Protected</h3>;


/*
export default function App() {
    return (
        <Router>
            <div>
                <nav>
                    <ul>
                        <li>
                            <Link to="/">Home</Link>
                        </li>
                        <li>
                            <Link to="/about">About</Link>
                        </li>
                        <li>
                            <Link to="/users">Users</Link>
                        </li>
                    </ul>
                </nav>

                <Switch>
                    <Route path="/about">
                        <About />
                    </Route>
                    <Route path="/users">
                        <Users />
                    </Route>
                    <Route path="/">
                        <Home />
                    </Route>
                </Switch>
            </div>
        </Router>
    );
}

function Home() {
    return  <div className='min-h-screen min-w-screen bg-gray-900 text-center text-6xl text-gray-200 grid items-center tracking-wider'>
        AWESOME BETTING APP
    </div>;
}

function About() {
    return <h2>About</h2>;
}

function Users() {
    return <h2>Users</h2>;
}
*/