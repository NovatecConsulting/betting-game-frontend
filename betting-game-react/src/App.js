import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import './App.css';
import Navbar from './components/Navbar/Navbar';

function App() {
  return (
    <div>
      <Router>
        <Navbar></Navbar>
      </Router>
    </div>
  );
}

export default App;
