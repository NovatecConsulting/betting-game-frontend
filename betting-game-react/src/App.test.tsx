import React from "react";
import { shallow } from "enzyme";
import { Navbar } from "./components/Navbar";
import App from "./App";

describe(App.name + " component", () => {
    it("should render and match snapshot.", () => {
        const wrapper = shallow(<App />);

        expect(wrapper.find(Navbar)).toExist();
        expect(wrapper).toMatchSnapshot();
    });
});
