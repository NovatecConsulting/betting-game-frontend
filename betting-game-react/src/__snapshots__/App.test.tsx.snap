// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should render and match snapshot. 1`] = `
<div>
  <BrowserRouter>
    <Navbar />
  </BrowserRouter>
</div>
`;
